### PR TITLE
[WIP] Add support for compiled inference and quantization

### PR DESCRIPTION
**Pull request by iammosespaulr**

### CODE REVIEW

File: surya/tables.py [hunk 1]
```diff
@@ -39,6 +39,19 @@ def sort_bboxes(bboxes, tolerance=1):
     return sorted_page_blocks
 
 
+def pad_to_batch_size(tensor: torch.Tensor, batch_size: int) -> torch.Tensor:
+    current_batch_size = tensor.shape[0]
+    if current_batch_size >= batch_size:
+        return tensor
+
+    pad_size = batch_size - current_batch_size
+    repeats = (pad_size + current_batch_size - 1) // current_batch_size
```
Comment (VikParuchuri) [inline surya/tables.py:48]: Why is this not zero padding?

**Review by VikParuchuri — COMMENTED** [review]
Looks mostly good, but I haven't had time to do manual testing

File: surya/tables.py (hunk 1 above)
Comment (iammosespaulr) [reply surya/tables.py:48]: That should work if we were just padding the image tensor but we've also got `batch_bbox_counts` which appears to be constructed in a specific way and breaks the model if it's just zero padded.
https://github.com/VikParuchuri/surya/blob/c33bba493ed3e4926e05f4332770ad51920a642d/surya/model/table_rec/processor.py#L246-L253. So it seemed simpler to do repeat padding. What do you think?

File: surya/model/layout/model.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import torch
+
+from surya.model.detection.model import EfficientViTForSemanticLayoutSegmentation
+from surya.model.detection.config import EfficientViTConfig
+from surya.model.detection.processor import SegformerImageProcessor
+from surya.settings import settings
+
+def load_model(checkpoint=settings.DETECTOR_MODEL_CHECKPOINT, device=settings.TORCH_DEVICE_MODEL, dtype=settings.MODEL_DTYPE) -> EfficientViTForSemanticLayoutSegmentation:
```
Comment (VikParuchuri) [inline surya/model/layout/model.py:8]: Please update the docs to indicate how to import the layout model - I think we also have to update the ocr app and cli scripts, right?  This also needs corresponding changes in marker, since it loads the layout model

Comment (iammosespaulr) [reply surya/model/layout/model.py]: Updated!
